packages/application/src/index.ts

```diff
@@ -146,7 +146,13 @@ class JupyterLab extends Application<ApplicationShell> {
     if (!Array.isArray(data)) {
       data = [data];
     }
-    data.forEach(item => { this.registerPlugin(item); });
+    data.forEach(item => {
+      try {
+        this.registerPlugin(item);
+      } catch (error) {
+        window.alert(error);
```

Comment:
Perhaps keep track of all startup errors and show them all at once, using `showDialog`?

Reply:
I have changed it to alert the user when the application loads. Here is an example when I load all extensions twice:

![](http://recordit.co/aXoltQDqzm.gif)